precompile/contracts/deployerallowlist/deployerallowlisttest/simulated_test.go

```diff
@@ -0,0 +1,387 @@
+			},
+		},
+		{
+			name: "should allow admin to add deployer via contract",
+			test: func(t *testing.T, backend *sim.Backend, allowList *allowlisttest.IAllowList) {
+				allowListTestAddr, allowListTest := deployAllowListTestContract(t, backend, admin)
+				otherContractAddr, _ := deployAllowListTestContract(t, backend, admin)
+
+				verifyRole(t, allowList, allowListTestAddr, allowlist.NoRole)
+				setAsAdmin(t, backend, allowList, admin, allowListTestAddr)
+				verifyRole(t, allowList, allowListTestAddr, allowlist.AdminRole)
+
+				tx, err := allowListTest.SetEnabled(admin, otherContractAddr)
+				require.NoError(t, err)
+				waitReceipt(t, backend, tx)
+
+				isEnabled, err := allowListTest.IsEnabled(nil, otherContractAddr)
+				require.NoError(t, err)
+				require.True(t, isEnabled)
+				verifyRole(t, allowList, otherContractAddr, allowlist.EnabledRole)
+			},
+		},
+		{
+			name: "should allow enabled address to deploy contracts",
+			test: func(t *testing.T, backend *sim.Backend, allowList *allowlisttest.IAllowList) {
+				allowListTestAddr, allowListTest := deployAllowListTestContract(t, backend, admin)
+				deployerContractAddr, deployerContract := deployAllowListTestContract(t, backend, admin)
+
+				setAsAdmin(t, backend, allowList, admin, allowListTestAddr)
+
+				tx, err := allowListTest.SetEnabled(admin, deployerContractAddr)
+				require.NoError(t, err)
+				waitReceipt(t, backend, tx)
+
+				isEnabled, err := allowListTest.IsEnabled(nil, deployerContractAddr)
+				require.NoError(t, err)
+				require.True(t, isEnabled)
+
+				tx, err = deployerContract.DeployContract(admin)
+				require.NoError(t, err)
+				receipt := waitReceipt(t, backend, tx)
+				require.Equal(t, types.ReceiptStatusSuccessful, receipt.Status)
+			},
+		},
+		{
+			name: "should allow admin to revoke deployer",
+			test: func(t *testing.T, backend *sim.Backend, allowList *allowlisttest.IAllowList) {
+				allowListTestAddr, allowListTest := deployAllowListTestContract(t, backend, admin)
+				deployerContractAddr, _ := deployAllowListTestContract(t, backend, admin)
+
+				setAsAdmin(t, backend, allowList, admin, allowListTestAddr)
+
+				tx, err := allowListTest.SetEnabled(admin, deployerContractAddr)
+				require.NoError(t, err)
+				waitReceipt(t, backend, tx)
+
+				isEnabled, err := allowListTest.IsEnabled(nil, deployerContractAddr)
+				require.NoError(t, err)
+				require.True(t, isEnabled)
+
+				tx, err = allowListTest.Revoke(admin, deployerContractAddr)
+				require.NoError(t, err)
+				waitReceipt(t, backend, tx)
+
+				verifyRole(t, allowList, deployerContractAddr, allowlist.NoRole)
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			backend := newBackendWithDeployerAllowList(t)
+			defer backend.Close()
+
+			allowList, err := allowlisttest.NewIAllowList(deployerallowlist.ContractAddress, backend.Client())
+			require.NoError(t, err)
+
+			tc.test(t, backend, allowList)
+		})
+	}
+}
```

Comment:
Everything up until this point is a part of https://github.com/ava-labs/subnet-evm/pull/1850 -- already approved